flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -215,6 +226,12 @@ public void run() {
 								refreshDir, jobID, iae);
 							continue;
 						}
+
+						if (++historySize > maxHistorySize) {
```

Comment:
I would suggest to increment `historySize` in a separate line. This is usually more robust because people might confuse `++historySize` and `historySize++` when refactoring the if condition in the future.

Reply:
Refactored accordingly.